app/services/track_length_service.rb

```diff
@@ -0,0 +1,20 @@
+require 'enumerator'
+class TrackLengthService
+  def initialize(track)
+    @track = track.sort_by(&:timestamp)
```

Comment:
i think `sort_by()` is a ruby method, where as `.order()` is going to tell active record to write sql: `order by`. I'd prefer to sql ordering over ruby sorting if possible